math/vecops/inc/ROOT/RVec.hxx

```diff
@@ -38,6 +38,28 @@
 #endif
 
 namespace ROOT {
+namespace VecOps {
+template<typename T>
+class RVec;
+}
+namespace Internal {
+namespace VecOps {
+
+// We use this helper to workaround a limitation of compilers such as
+// gcc 4.8 amd clang on osx 10.14 for which std::vector<bool>::emplace_back
+// is not defined.
+// We have here the template for all types while at the bottom of this file
+// the partial specialisation for T == bool, which uses push_back.
+template<typename T, typename... Args>
+struct REmplaceBackHelper{
+static void EmplaceBack(typename ::ROOT::VecOps::RVec<T>::Impl_t &v, Args &&... args)
```

Comment:
You can replace `::ROOT::VecOps::RVec<T>::Impl_t` by `T` and make the specialization directly with `std::vector<bool>&`, then you don't need to forward declare `RVec<T>` and you can put both implementation together at the top of the file.
Or does this have any implications, which I don't see?

Reply:
expose explicitly the implementation of the storage of RVec, which is a vector<T, RAdoptAllocator<T>>

Reply:
I take it back. Nice proposal. Let me see what I can do.